components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/JavaMemoryUsageLogger.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.healthcheck.api.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.lang.management.*;
+import java.util.EnumMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Service which logs the memory usage of carbon server.
+ */
+public class JavaMemoryUsageLogger {
+
+    static int BYTES_PER_MB = 1024 * 1000;
+    private static MemoryMXBean memoryMXBean;
+
+    private static final Log log = LogFactory.getLog(JavaMemoryUsageLogger.class);
+    int wait = Constants.MemoryUsageLoggerConfig.DEFAULT_WAIT_SECONDS;
+    int interval = Constants.MemoryUsageLoggerConfig.DEFAULT_INTERVAL_SECONDS;
+    ScheduledExecutorService ses;
```

Comment:
Private ?

Reply:
also need a better variable name describing the purpose

Reply:
Fixed